File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    inorderTraversal(root) {
        // inorder traversal says what 
         const result = [] ; 

         function dfs(node) {
            if  (!node) return ; 
             dfs(node.left) ; 
             result.push(node.val); 
             dfs(node.right) ; 
         }
         dfs(root) ; 
         return result ; 
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        diameter = 0 

        def dfs(node) : 
            nonlocal diameter 

            if node is None : 
                return -1 
            
            leftHeight = dfs(node.left)
            rightHeight = dfs(node.right)

            diameter = max ( diameter , leftHeight + rightHeight + 2)

            return 1 + max(leftHeight , rightHeight)

        dfs(root)

        return diameter 
            
        
        